### PR TITLE
Translations/ Connect hints

### PR DESCRIPTION
**Pull request by eadams17**
## WHAT
add translations for hints in Connect

## WHY
this is part of the larger translation project for translating Connect/Grammar activities

## HOW
hook into `concept_feedbacks` API, populate data in frontend, conditionally render concept explanation if translation is present

note: we use language for question data and locale for concept feedback in the backend currently-- we should probably standardize what argument is passed

### Screenshots
(If applicable. Also, please censor any sensitive data)
Screen Shot 2024-09-12 at 3 10 10 PM

### Notion Card Links
(Please provide links to any relevant Notion card(s) relevant to this PR.)
https://www.notion.so/quill/If-you-pick-a-language-show-the-hint-descriptions-in-that-language-your-second-language-25b0fd9142974fcca4c2c2c460101067?pvs=4
https://www.notion.so/quill/Translations-changes-6dc4b121ceb94109bcd13c07c913ad96?pvs=4

### What have you done to QA this feature?
(Provide enough detail that a reviewer could assess whether additional QA should be done. For larger projects, additionally use the Engineer Feature Testing Notion template. Review Guidelines if needed: https://www.notion.so/quill/Github-PR-QA-Guidelines-49e99fc965654ceeb8c6249bd9d181d7)
tested locally and on staging with Rachel to ensure that various different activities display translated hints as expected

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? | yes
Have you deployed to Staging? | yes
Self-Review: Have you done an initial self-review of the code below on Github? | yes

### CODE REVIEW

File: services/QuillLMS/client/app/bundles/Shared/components/translations/languagePicker.tsx [hunk 1]
```diff
@@ -10,8 +10,9 @@ interface LanguagePickerProps {
 }
 
 const LanguagePicker = ({ language, updateLanguage, languageOptions }: LanguagePickerProps) => {
+  const capitalizedLanguageOptions = languageOptions ? languageOptions.map(option => ({ value: option.value, label: titleCase(option.label) })) : null
   const options = (): Array<{value: string, label: string}> =>
-    languageOptions ?? defaultLanguages.map(language => ({
+    capitalizedLanguageOptions ?? defaultLanguages.map(language => ({
```
Comment (anathomical) [inline services/QuillLMS/client/app/bundles/Shared/components/translations/languagePicker.tsx:15]: I wasn't familiar with `??`.  I like its use here.

**Review by anathomical — APPROVED** [review]

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx [hunk 2]
```diff
@@ -244,35 +244,43 @@ export class PlayFillInTheBlankQuestion extends React.Component<PlayFillInTheBla
   }
 
   renderConceptExplanation = () => {
-    const { conceptsFeedback, question } = this.props
+    const { conceptsFeedback, question, showTranslation } = this.props
     // TODO: update Response interface in quill-marking-logic
     const latestAttempt: Attempt|undefined = getLatestAttempt(question.attempts);
     if (latestAttempt && latestAttempt.response && !latestAttempt.response.optimal ) {
```
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx]: A guard clause here will simplify the large conditional:
```  if (!latestAttempt?.response || latestAttempt.response.optimal) { return null; }```

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx
Comment (brendanshean) [inline services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx:1]: I think there's some refactoring that can be done with the different branches.

I realize there's some older code in here, but maybe some refactoring here.  Something like:

```
  const getTranslatedData = (key) => 
    showTranslation && conceptsFeedback?.translated_data 
      ? conceptsFeedback.translated_data[key] 
      : null;

  const renderExplanation = (data, key) => {
    if (!data) return null;
    const translatedData = getTranslatedData(key);
    return <ConceptExplanation {...data} translatedExplanation={translatedData} />;
  };
```

Then you have something like:

```
if (latestAttempt.response.conceptResults) {
  const conceptID = this.getNegativeConceptResultForResponse(latestAttempt.response.conceptResults);
  if (conceptID) {
    const key = conceptID.conceptUID;
    return renderExplanation(conceptsFeedback.data[key], key);
  }
} else if (latestAttempt?.response?.concept_results) {
  const conceptID = this.getNegativeConceptResultForResponse(latestAttempt.response.concept_results);
  if (conceptID) {
    const key = conceptID.conceptUID;
    return renderExplanation(conceptsFeedback.data[key], key);
  }
} else if (question?.modelConceptUID) {
  const key = question.modelConceptUID;
  return renderExplanation(conceptsFeedback.data[key], key);
} else if (question?.conceptID) {
  const key = question.conceptID;
  return renderExplanation(conceptsFeedback.data[key], key);
}
```

**Review by brendanshean — APPROVED** [review]
Nice work on putting this together.  This is an exciting feature.

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx (hunk 2 above)
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx]: good call, I'll update

File: services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx
Comment (eadams17) [reply services/QuillLMS/client/app/bundles/Connect/components/studentLessons/fillInBlank.tsx:1]: I'll do a bit of cleaning up of the older code here

**Pull request merged**